docs/sdk/migrations/6.0.0.md

```diff
@@ -0,0 +1,491 @@
+  authMethods: [...params.authMethods],
+  pkpPublicKey: params.pkpPublicKey,
+  expiration: props.expiration,
+  resources: props.resources,
+  chainId: 1,
+  resourceAbilityRequests: props.resourceAbilityRequests,
+});
+
+// v6
+const response = await this.signSessionKey({
+  sessionKey: props.sessionKey,
+  statement: props.statement || 'Some custom statement.',
+  authMethods: [...params.authMethods],
+  pkpPublicKey: params.pkpPublicKey,
+  expiration: props.expiration,
+  resources: props.resources,
+  chainId: 1,
+  resourceAbilityRequests: props.resourceAbilityRequests,
+
+  // -- optional fields
+  ...(props.litActionCode && { litActionCode: props.litActionCode }),
+  ...(props.ipfsId && { ipfsId: props.ipfsId }),
+  ...(props.jsParams && { jsParams: props.jsParams }),
+});
+```
+
+#### getPkpSessionSigs
+
+Rather than creating a callback for the `authNeededCallback` parameter in the `getSessionSigs` function, this function creates the callback and signs the session key under the hood using the `signSessionKey` function.
+
+```tsx
+import { EthWalletProvider } from '@lit-protocol/lit-auth-client';
+
+// -- preparing the parameters
+const authMethod = await EthWalletProvider.authenticate({
+  signer: YOUR_WALLET_SIGNER,
+  litNodeClient,
+});
+
+const authMethodOwnedPkpPublicKey = '0x..';
+
+const resourceAbilityRequests = [
+  {
+    resource: new LitPKPResource('*'),
+    ability: LitAbility.PKPSigning,
+  },
+  {
+    resource: new LitActionResource('*'),
+    ability: LitAbility.LitActionExecution,
+  },
+];
+
+// -- get pkp session sigs
+const pkpSessionSigs = await litNodeClient.getPkpSessionSigs({
+  pkpPublicKey: authMethodOwnedPkpPublicKey,
+  authMethods: [authMethod],
+  resourceAbilityRequests: resourceAbilityRequests,
+});
+```
+
+##### Custom Authentication
+
+For custom authentication, you can set custom Javascript code that would be executed on Lit Action. 
+
+```tsx
+import { EthWalletProvider } from '@lit-protocol/lit-auth-client';
+
+// -- preparing the parameters
+const authMethod = await EthWalletProvider.authenticate({
+  signer: YOUR_WALLET_SIGNER,
+  litNodeClient,
+});
+
+const authMethodOwnedPkpPublicKey = '0x..';
+
+const resourceAbilityRequests = [
+  {
+    resource: new LitPKPResource('*'),
+    ability: LitAbility.PKPSigning,
+  },
+  {
+    resource: new LitActionResource('*'),
+    ability: LitAbility.LitActionExecution,
+  },
+];
+
+// In a browser environment, instead of using Node.js's Buffer class 
+// for handling binary data and encoding it to base64, 
+// you can use the TextEncoder API to convert string data to a 
+// Uint8Array and then convert it to a base64 string using btoa() function.
+const customAuthLitActionCode = Buffer.from(`
+// Works with an AuthSig AuthMethod
+if (Lit.Auth.authMethodContexts.some(e => e.authMethodType === 1)) {
+  LitActions.setResponse({ response: "true" });
+} else {
+  LitActions.setResponse({ response: "false" });
+}
+`).toString('base64');
```

Comment:
Can we just switch this to `TextEncoder` NodeJS also supports it so no reason to use `Buffer.from`